src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,792 @@
+/// \param[in] _default Default value of type T
+/// \return Value from `_vec` at `_index`, or `_default` if `_vec` is not large enough
+template <typename T>
+T GetNthOrDefault(const std::vector<T> &_vec,
+                  const size_t &_index,
+                  const T &_default)
+{
+  if (_index < _vec.size())
+  {
+    return _vec[_index];
+  }
+  else
+  {
+    return _default;
+  }
+}
+
+/////////////////////////////////
+/// JointTrajectoryController ///
+/////////////////////////////////
+
+JointTrajectoryController::JointTrajectoryController()
+    : dataPtr(std::make_unique<JointTrajectoryControllerPrivate>())
+{
+}
+
+void JointTrajectoryController::Configure(const Entity &_entity,
+                                          const std::shared_ptr<const sdf::Element> &_sdf,
+                                          EntityComponentManager &_ecm,
+                                          EventManager & /*_eventManager*/)
+{
+  // Make sure the controller is attached to a valid model
+  const auto model = Model(_entity);
+  if (!model.Valid(_ecm))
+  {
+    ignerr << "[JointTrajectoryController] Failed to initialize because [" << model.Name(_ecm)
+           << "(Entity=" << _entity << ")] is not a model. Please make sure that"
+                                       " JointTrajectoryController is attached to a valid model.\n";
+    return;
+  }
+  ignmsg << "[JointTrajectoryController] Setting up controller for [" << model.Name(_ecm)
+         << "(Entity=" << _entity << ")].\n";
+
+  // Get list of enabled joints
+  // If empty, enable all joints
+  const auto enabledJoints = ParseVectorParam<std::string>(_sdf->Get<std::string>("joint_names"));
+
+  // Iterate over all joints under the model entity and configure them
+  for (const auto &jointEntity : _ecm.ChildrenByComponents(_entity, components::Joint()))
+  {
+    this->dataPtr->ConfigureJoint(jointEntity, _sdf, _ecm, enabledJoints);
+  }
+
+  // Make sure at least one joint is configured
+  if (this->dataPtr->actuatedJoints.empty())
+  {
+    ignerr << "[JointTrajectoryController] Failed to initialize because [" << model.Name(_ecm)
+           << "(Entity=" << _entity
+           << ")] has no supported joints.\n";
+    return;
+  }
+
+  // Get additional parameters from SDF
+  if (_sdf->HasAttribute("use_header_start_time"))
+  {
+    this->dataPtr->useHeaderStartTime = _sdf->Get<bool>("use_header_start_time");
+  }
+  else
+  {
+    this->dataPtr->useHeaderStartTime = false;
+  }
+
+  // Subscribe to joint trajectory commands
+  auto trajectoryTopic = _sdf->Get<std::string>("topic");
+  if (trajectoryTopic.empty())
+  {
+    trajectoryTopic = "/model/" + model.Name(_ecm) + "/joint_trajectory";
+    ignmsg << "[JointTrajectoryController] No topic specified for joint trajectories,"
+              " defaulting to ["
+           << trajectoryTopic << "].\n";
+  }
+  else
+  {
+    if (trajectoryTopic[0] != '/')
+    {
+      trajectoryTopic.insert(0, "/");
+    }
```

Comment:
Is this check necessary? I think Ignition Transport automatically adds the `/`

Reply:
Removed as part of https://github.com/ignitionrobotics/ign-gazebo/commit/8feb9c49f0262a59809f854e9c817768f0b01d61